.github/workflows/nodejs-macos.yml

```
@@ -22,3 +22,14 @@ jobs:
         uses: ./.github/actions/prepare
       - name: Build & Test
         uses: ./.github/actions/test
+  re-run:
+    name: Rerun flaky macOS tests
+    needs: compatibility-test-macos
+    if: failure() && fromJSON(github.run_attempt) < 3
```

```suggestion
    if: needs.compatibility-test-macos.result == 'failure' && fromJSON(github.run_attempt) < 3
```

(so not checking if any prev job failed)